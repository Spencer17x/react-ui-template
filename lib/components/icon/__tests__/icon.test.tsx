import React from "react";
import { render } from "enzyme";
import Icon from "..";
import mountTest from "../../../../tests/shared/mountTest";

describe("render", () => {
  mountTest(() => <Icon type="add" />);
  it("render correctly", function() {
    const wrapper = render(<Icon type="add" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("render loading", function() {
    const wrapper = render(
      <Icon type="loading" spin={true} fill="#1890ff" interval={1200} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("props", () => {
  it("props fill", function() {
    const wrapper = render(<Icon type="add" fill="green" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("props spin", function() {
    const wrapper = render(<Icon type="add" spin={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("props interval", function() {
    const wrapper = render(<Icon type="add" interval={500} />);
    expect(wrapper).toMatchSnapshot();
  });
});
