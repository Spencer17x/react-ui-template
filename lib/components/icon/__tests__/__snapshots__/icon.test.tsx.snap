// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props props fill 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "class": "f-icon",
      "fill": "green",
      "style": "animation-duration:0ms",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "#add",
        },
        "children": Array [],
        "name": "use",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": "http://www.w3.org/1999/xlink",
        },
        "x-attribsPrefix": Object {
          "href": "xlink",
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "fill": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "fill": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`props props interval 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "class": "f-icon f-icon-spin",
      "style": "animation-duration:500ms",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "#add",
        },
        "children": Array [],
        "name": "use",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": "http://www.w3.org/1999/xlink",
        },
        "x-attribsPrefix": Object {
          "href": "xlink",
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`props props spin 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "class": "f-icon f-icon-spin",
      "style": "animation-duration:1800ms",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "#add",
        },
        "children": Array [],
        "name": "use",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": "http://www.w3.org/1999/xlink",
        },
        "x-attribsPrefix": Object {
          "href": "xlink",
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`render render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "class": "f-icon",
      "style": "animation-duration:0ms",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "#add",
        },
        "children": Array [],
        "name": "use",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": "http://www.w3.org/1999/xlink",
        },
        "x-attribsPrefix": Object {
          "href": "xlink",
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`render render loading 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-hidden": "true",
      "class": "f-icon f-icon-spin",
      "fill": "#1890ff",
      "style": "animation-duration:1200ms",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "#loading",
        },
        "children": Array [],
        "name": "use",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": "http://www.w3.org/1999/xlink",
        },
        "x-attribsPrefix": Object {
          "href": "xlink",
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "fill": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-hidden": undefined,
      "class": undefined,
      "fill": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
